include/drivers/irq/am335x.h

```diff
@@ -119,8 +119,20 @@ static inline void handleSpuriousIRQ(void)
     dsb();
 }
 
+void setIRQTrigger(irq_t irq, bool_t trigger)
+{
+    /* Unreachable; not supported on this platform. */
+    halt();
```

Comment:
Why the halt though? That just increases the binary size for no reason.

Reply:
It forces me to prove that the function is unreachable.

Reply:
(We could also just make it empty, now that I've proved it, but if we ever change anything we won't be forced to prove that it's not reachable)

Reply:
Somehow @midnightveil's comment only popped up after I had replied -- the UNREACHABLE macro is only for the compiler, really, although staring at it now, maybe we could give it a spec that also forces us to prove that it is unreachable. I have the feeling things would already break if it was reachable, because it doesn't have a body, but making it explicit could be useful.

Reply:
That sound like a better, more explicit approach than sprinkling `halt` everywhere.

This reminds me I should read commit messages more during review. I usually do, but not for quick reviews like this one.

Reply:
Nice!